src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -1928,6 +1945,163 @@ real64 CompositionalMultiphaseBase::setNextDtBasedOnStateChange( real64 const &
   return std::min( std::min( nextDtPressure, nextDtPhaseVolFrac ), nextDtTemperature );
 }
 
+real64 CompositionalMultiphaseBase::setNextDtBasedOnCFL( const geos::real64 & currentDt, geos::DomainPartition & domain )
```

Comment:
this duplicates a lot from CompositionalMultiphaseStatistics::computeCFLNumbers, right? any way to share the code between the two?

Reply:
Will do some clean up

Reply:
looks good, thanks for cleaning it up